tests/test_content_client.py

```diff
@@ -0,0 +1,169 @@
+import pytest
+import testinfra
+import time
+import subprocess
+
+
+@pytest.fixture
+def katello_client(request, entities, registration_hostname):
+    image_id = subprocess.check_output(['podman', 'images', '-q', '--filter', f"reference={request.param}"])
+    assert image_id
+
+    try:
+        container_id = subprocess.check_output([
+            'podman',
+            'run',
+            '--volume',
+            '/dev/log:/dev/log',
+            '--detach',
+            '--tty',
+            request.param
+        ]).decode().strip()
+        container = testinfra.get_host("podman://" + container_id)
+
+        container.run_expect([0], f"rpm -Uvh http://{registration_hostname}/pub/katello-ca-consumer-latest.noarch.rpm")
+        container.run_expect([0], f"subscription-manager register --force --org={entities['organization_label']} --activationkey=\"{entities['activation_key']}\"")
+
+        consumed = container.check_output("subscription-manager list --consumed")
+        assert entities['product'] in consumed
+
+        # return a testinfra connection to the container
+        yield container
+
+    finally:
+        subprocess.check_call(['podman', 'rm', '-f', container_id])
+
+
+@pytest.fixture
+def katello_agent_client(katello_client, os_release):
+    if os_release not in ("6", "7"):
+        pytest.skip(f"katello-agent not available on EL{os_release}")
+
+    katello_client.run_expect([0], 'yum -y install katello-agent')
+
+    katello_agent = katello_client.package('katello-agent')
+    assert katello_agent.is_installed
+
+    return katello_client
+
+
+@pytest.fixture
+def os_release(katello_client):
```

Comment:
I'm sure there will come a time when we need to differentiate on major.minor. Let's be explicit that it's only the major version.
```suggestion
def os_release_major(katello_client):
```